### PR TITLE
Use polling instead of filters to wait for tx confirmation

### LINKED ISSUE #533: Ethcontract erroneously waits for transaction even after it was mined
From slack discussion:

@taminomara:
> I've noticed a couple of times when executing methods via ethcontract that they may hang and never resolve even though transaction gets mined successfully (I'm using local chain running in dev mode).

@nlordell:
> So, what happens is that:
> - Transaction is sent by ethcontract
> - Block number is retrieved and is block X
> - transaction gets mined and is now block X+1
> - Transaction receipt gets retrieved with block number X+1
> - ethcontract thinks that it needs to wait for 1 block
> - creates a eth_newBlockFilter to get notifications about new blocks
> - waits for ever to get notified about new blocks, but they don't come because Geth only mines when a new transcation comes is.

@taminomara:
> Interesting. I wonder why we didn't catch it sooner? I guess geth mines some additional blocks?

@nlordell:
> I think its because:
> - On Ganache, the transaction gets mined immediately, so the race is not possible
> - On real nodes, new blocks eventually come in, so worst case we wait an extra block

### PR DESCRIPTION
**Pull request by taminomara**
There's a race condition. If transaction gets mined after we fetch latest block number, but before we create a filter, then we wait for more blocks than necessary. If new blocks are not generated (i.e. dev setup or testchains with low activity), waiting for transaction may hang.

We attempt to remedy this issue by switching entirely to polling for block numbers. In theory, this should also decrease node's load. A potential downside is that confirmation times may increase. That is, we now poll node every seven seconds instead of confirming transaction immediately after the target block was mined. We will fix that later by introducing exponential backoff to polling mechanism — see #534.

Fix #533.

### CODE REVIEW

File: ethcontract/src/transaction/confirm.rs
Comment (nlordell) [inline ethcontract/src/transaction/confirm.rs]: Nice! This will solve the race we saw.

**Review by nlordell — COMMENTED** [review]
Looks good so far! I like how it cut down on complexity.

**Review by vkgnosis — APPROVED** [review]
Looks good to me but let's wait for Nick's approval before merging.

**Comment by taminomara** [comment]
For context, openethereum team confirmed that polling generates less load for node than filters ([slack thread]).

[slack thread]: https://gnosisinc.slack.com/archives/C0107GMHWKY/p1623675227061700

**Comment by vkgnosis** [comment]
Also in practice at least with our load balanced set ups filters not great because every request might go to a new node which doesn't have the filter set up yet.

**Review by nlordell — APPROVED** [review]

**Pull request merged**
